language/move-binary-format/src/file_format_common.rs

```diff
@@ -405,8 +405,12 @@ pub const VERSION_5: u32 = 5;
 ///  + u16, u32, u256 integers and corresponding Ld, Cast bytecodes
 pub const VERSION_6: u32 = 6;
 
+/// Version 7: changes compared with version 6
+/// Fixed bug in signature checking pass of verifier--did not do ability checks on the `T` in `&T`, `&mut T`, or `vector<T>`
+pub const VERSION_7: u32 = 7;
+
 // Mark which version is the latest version
-pub const VERSION_MAX: u32 = VERSION_6;
+pub const VERSION_MAX: u32 = VERSION_7;
```

Comment:
@vgao1996 @tnowacki: I assume it's too late to get this into version 6, but let me know if not

Reply:
We would love to have this in v6 but it basically needs to land ASAP.

Reply:
Though I'm wondering: why would this be a file format change?

We have adapted the general policy that restricting bytecode from doing crazy things is _not_ a breaking change. This hurts us in replay, as @runtian-zhou knows, but it seems rather unpractical to allow old 'malicious/broken/...' code to continue to exist. Runtian has build a tool which scans our published modules for whether they are affected, and we try to get an idea what we are breaking. Our policy is if you earlier successfully published a module and we figure it is a danger for our network, we do not continue to allow it.

Reply:
> We have adapted the general policy that restricting bytecode from doing crazy things is not a breaking change.

This principle makes sense to me. My preference would also be to get rid of the version gating--will fix/land.